Sources/Features/Network/MultipartRequest.swift

```diff
@@ -48,41 +48,85 @@ extension MultipartRequest {
         var multipartRequest = urlRequest
         let boundary = generateBoundaryString()
         multipartRequest.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
-
-        let boundaryPrefix = "--\(boundary)\r\n"
-        let body = NSMutableData()
-        // attributes
-        for attribute in report.attributes {
-            body.appendString(boundaryPrefix)
-            body.appendString("Content-Disposition: form-data; name=\"\(attribute.key)\"\r\n\r\n")
-            body.appendString("\(attribute.value)\r\n")
-        }
-        // report file
-        body.appendString(boundaryPrefix)
-        body.appendString("Content-Disposition: form-data; name=\"upload_file\"; filename=\"upload_file\"\r\n")
-        body.appendString("Content-Type: application/octet-stream\r\n\r\n")
-        body.append(report.reportData)
-        body.appendString("\r\n")
-
-        // attachments
-        for attachment in Set(report.attachmentPaths).compactMap(Attachment.init(filePath:)) {
-            body.appendString(boundaryPrefix)
-            body.appendString("Content-Disposition: form-data; name=\"\(attachment.filename)\"; filename=\"\(attachment.filename)\"\r\n")
-            body.appendString("Content-Type: \(attachment.mimeType)\r\n\r\n")
-            body.append(attachment.data)
-            body.appendString("\r\n")
+        
+        // temporary file to stream data
+        let tempURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(UUID().uuidString)
+        
+        do {
+            let fileCreated = FileManager.default.createFile(atPath: tempURL.path, contents: nil, attributes: nil)
+            if !fileCreated {
+                throw HttpError.fileCreationFailed(tempURL)
+            }
+            
+            let fileHandle = try FileHandle(forWritingTo: tempURL)
+            defer {
+                if #available(iOS 13.0, tvOS 13.0, macOS 11.0, *) {
+                    try? fileHandle.close()
+                } else {
+                    fileHandle.closeFile()
+                }
+            }
+            
+            // attributes
+            var attributesString = ""
+            for attribute in report.attributes {
+                attributesString += "--\(boundary)\r\n"
+                attributesString += "Content-Disposition: form-data; name=\"\(attribute.key)\"\r\n\r\n"
+                attributesString += "\(attribute.value)\r\n"
+            }
+            try writeToFile(fileHandle, attributesString)
+            
+            // report
+            var reportString = "--\(boundary)\r\n"
```

Comment:
It would be better if we start with report rather than with attributes - if the application stops during submission part, we can still submit a valid report.